src/interface/src/app/services/redirect.service.ts

```diff
@@ -1,19 +1,22 @@
 import { Injectable } from '@angular/core';
+import { LoginRedirectStorageService } from '@services/local-storage.service';
 
 export interface RedirectData {
   url: string;
-  userHash?: string;
+  userHash: string | null;
 }
 
 @Injectable({
   providedIn: 'root',
 })
 export class RedirectService {
-  readonly key = 'loginRedirect';
+  constructor(
+    private loginRedirectStorageService: LoginRedirectStorageService
+  ) {}
 
   setRedirect(url: string, user?: string) {
     const userHash = user ? Buffer.from(user).toString('base64') : null;
-    localStorage.setItem(this.key, JSON.stringify({ url, userHash }));
+    this.loginRedirectStorageService.setItem({ url, userHash });
```

Comment:
Did this one just to test the implementation. If these changes look good, going to implement them on other places that deal with localstorage